src/java/org/apache/cassandra/cql3/statements/schema/AlterTableStatement.java

```diff
@@ -93,7 +94,12 @@ public Keyspaces apply(Keyspaces schema) throws UnknownHostException
                             : keyspace.getTableOrViewNullable(tableName);
 
         if (null == table)
+        {
+            if (ifTableExists)
+                return schema;
```

Comment:
Looks like when using the above statement, i.e not returning `schema` would throw an exception when we try to alter a table that doesn't exist using `IF EXISTS`. For e.g consider this test case 
```
@Test
    public void testAlterTableWithoutCreateTableWithIfExists() throws Throwable
    {
        String tbl1 = KEYSPACE + "." + createTableName();
        assertNull(execute(String.format("ALTER TABLE IF EXISTS %s ADD myCollection list<text>;", tbl1)));
    }
```
would lead to 
```
java.lang.NullPointerException
	at org.apache.cassandra.cql3.statements.schema.AlterTableStatement.apply(AlterTableStatement.java:103)
```

Reply:
What I meant was:

```
if (!ifExists)
    throw ire("Table '%s.%s' doesn't exist", keyspaceName, tableName);

return schema; 
```
but your initial solution is also fine.

Reply:
thx. IMO, the one you have is cleaner, so will go with it.